mage_ai/cluster_manager/workspace/base.py

```diff
@@ -0,0 +1,123 @@
+import os
+import uuid
+from typing import Dict, Optional
+
+import yaml
+
+from mage_ai.cluster_manager.config import LifecycleConfig
+from mage_ai.cluster_manager.constants import ClusterType
+from mage_ai.cluster_manager.errors import WorkspaceExistsError
+from mage_ai.data_preparation.repo_manager import ProjectType, get_project_type
+from mage_ai.orchestration.constants import Entity
+from mage_ai.orchestration.db.models.oauth import User
+from mage_ai.settings.repo import get_repo_path
+
+
+class classproperty(property):
+    def __get__(self, owner_self, owner_cls):
+        return self.fget(owner_cls)
+
+
+class Workspace:
+    def __init__(self, name: str):
+        self.name = name
+
+    @classproperty
+    def project_folder(cls) -> str:
+        return os.path.join(get_repo_path(), 'projects')
+
+    @property
+    def config_path(self) -> str:
+        return os.path.join(self.project_folder, f'{self.name}.yaml')
+
+    @property
+    def config(self) -> Dict:
+        if os.path.exists(self.config_path):
+            with open(self.config_path, 'r', encoding='utf-8') as f:
+                return yaml.full_load(f)
+        return dict()
+
+    @property
+    def lifecycle_config(self) -> LifecycleConfig:
+        return LifecycleConfig.load(config=self.config.get('lifecycle_config', {}))
+
+    @property
+    def project_uuid(self) -> Optional[str]:
+        self.config.get('project_uuid')
+
+    def get_access(self, user: User) -> int:
+        return user.get_access(Entity.PROJECT, self.project_uuid)
+
+    @classmethod
+    def workspace_class_from_type(self, cluster_type: ClusterType) -> 'Workspace':
+        from mage_ai.cluster_manager.workspace.cloud_run import CloudRunWorkspace
```

Comment:
I would suggest put the import statement in its corresponding if else branch in case certain dependency packages are only installed for its cloud environment